File: app/src/main/java/viewmodels/ViewModelFactory.java
package viewmodels;

import android.app.Application;
import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.arch.lifecycle.ViewModelProviders;
import android.arch.lifecycle.ViewModelStoreOwner;

import data.DataRepository.repositorymodel.Country;

public class ViewModelFactory implements ViewModelProvider.Factory {

    private Application mApplication;
    private Country mParam;


    public ViewModelFactory(Application application, Country param){
        mApplication = application;
        mParam = param;
    }

    @Override
    public <T extends ViewModel> T create(Class<T> modelClass){
        return (T) new NetworkDataViewModel(mApplication,mParam);

    }



}

File: app/src/main/java/com/example/newsapp/viewmodels/ViewModelFactory.kt
package com.example.newsapp.viewmodels

import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider
import java.lang.RuntimeException
import javax.inject.Inject

@Suppress("UNCHECKED_CAST")
class ViewModelFactory @Inject constructor(private val mNewsViewModel: NewsViewModel) : ViewModelProvider.Factory {


    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        val viewModel: ViewModel
        if (modelClass == NewsViewModel::class.java) {
            viewModel = mNewsViewModel
        } else {
            throw RuntimeException("unsupported view model class: $modelClass")
        }
        return viewModel as T
    }


}